ppa/archive/models.py

```diff
@@ -389,13 +389,24 @@ class DigitizedWork(TrackChangesModel, ModelIndexable):
 
     class Meta:
         ordering = ("sort_title",)
+        # require unique combination of source id + page range,
+        # since we need to allow multiple excerpts from the same source
+        constraints = [
+            models.UniqueConstraint(
+                fields=["source_id", "pages_digital"], name="unique_sourceid_pagerange"
+            )
+        ]
 
     def get_absolute_url(self):
         """
         Return object's url for
         :class:`ppa.archive.views.DigitizedWorkDetailView`
         """
-        return reverse("archive:detail", kwargs={"source_id": self.source_id})
+        url_opts = {"source_id": self.source_id}
+        # start page must be specified if set but must not be included if empty
+        if self.pages_digital:
+            url_opts["start_page"] = self.first_page()
+        return reverse("archive:detail", kwargs=url_opts)
 
     def __str__(self):
         """Default string display. Uses :attr:`source_id`"""
```

Comment:
should this incorporate the page range if set? not a big deal, just seemed like it might make sense

Reply:
yes, good idea